packages/block-editor/src/components/rich-text/index.native.js

```diff
@@ -445,6 +448,27 @@ function RichTextWrapper(
 				isURL( plainText.trim() )
 			) {
 				mode = 'BLOCKS';
+				// Embed handler
+				embedHandlerPickerRef.current?.presentPicker( {
+					createEmbed: () => {
+						const content = pasteHandler( {
+							HTML: html,
+							plainText,
+							mode,
+							tagName,
+							preserveWhiteSpace,
+						} );
+
+						if ( content.length > 0 ) {
```

Comment:
Similar to the previous comment. This logic works but it is also repeated below. Based on how the `pasteHandler` operates I am assuming the `onReplace` logic [below](https://github.com/WordPress/gutenberg/pull/35204/files#diff-1bf4290b2ff8a9f2d347d3450576c98613f26f26f4ce05b850cf232483f7b28cR499) would work for other content but I haven't verified that as yet.